src/client/components/string-filter-menu/string-filter-menu.tsx

```diff
@@ -243,12 +244,12 @@ export class StringFilterMenu extends React.Component<StringFilterMenuProps, Str
       if (searchText) {
         var searchTextLower = searchText.toLowerCase();
         rowData = rowData.filter((d) => {
-          return String(d[SEGMENT]).toLowerCase().indexOf(searchTextLower) !== -1;
+          return String(d[dimension.name]).toLowerCase().indexOf(searchTextLower) !== -1;
         });
       }
 
       rows = rowData.map((d) => {
-        var segmentValue = d[SEGMENT];
+        var segmentValue = d[dimension.name];
```

Comment:
in general code that runs in a map or a filter rather than writing `var segmentValue = d[dimension.name];` I think it is nicer to write `var dimensionName = dimension.name` outside and then `var segmentValue = d[dimensionName];` inside. Saves the repeated key lookups.

This is cool though, just do not be surprised if I change this next time I touch this code ;-)